Bottom: 37cced031474387b542daeb7ccda117ddd2ee9be
Top:    a7860f04a856dd41d346e747727bd2dfd6a0017d
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-26 18:13:45 +0800

Refresh of debug

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index ec03caf..2641da6 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1778,9 +1778,10 @@ static ssize_t tun_get_user(struct tun_struct *tun, struct tun_file *tfile,
 	if ((tun->flags & TUN_TYPE_MASK) == IFF_TAP) {
 		align += NET_IP_ALIGN;
 		if (unlikely(len < ETH_HLEN ||
-				(gso.hdr_len && tun16_to_cpu(tun,
-		gso.hdr_len) < ETH_HLEN))){
-			printk("tun hdr check!\n");
+				(gso.hdr_len && tun16_to_cpu(tun, gso.hdr_len) < ETH_HLEN))){
+			printk("tun hdr check len %d "
+				"gso.hdr_len %d ETH_HLEN %d!\n", len,
+				tun16_to_cpu(tun, gso.hdr_len), ETH_HLEN);
 			return -EINVAL;
 		}
 	}
